dotcom-rendering/src/components/StickyBottomBanner/ReaderRevenueBanner.tsx

```diff
@@ -31,6 +31,7 @@ import { setAutomat } from '../../lib/setAutomat';
 import { useIsInView } from '../../lib/useIsInView';
 import { useOnce } from '../../lib/useOnce';
 import type { TagType } from '../../types/tag';
+import { hasRequiredConsents } from '../SignInGate/displayRule';
```

Comment:
Let's not reuse code from the SignInGate for this.
Given how simple `hasRequiredConsents` is, I think it's ok to duplicate that code in this file

Reply:
Sorted! Should we add this type of thing into a shared file?

Reply:
Generally the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) is a good pattern to follow. Two instances do not quite meet threshold for abstracting away :)